apps/air-discount-scheme/backend/src/app/modules/flight/flight.controller.ts

```diff
@@ -86,23 +222,54 @@ export class PublicFlightController {
       throw new ForbiddenException('User postalcode does not meet conditions')
     }
 
-    const {
-      unused: flightLegsLeft,
-    } = await this.flightService.countThisYearsFlightLegsByNationalId(
-      discount.nationalId,
+    let connectingFlight = false
+    let connectingId = undefined
+
+    const hasReykjavik = flight.flightLegs.some(
+      (flightLeg) =>
+        REYKJAVIK_FLIGHT_CODES.includes(flightLeg.origin) ||
+        REYKJAVIK_FLIGHT_CODES.includes(flightLeg.destination),
+    )
+
+    const hasAkureyri = flight.flightLegs.some(
+      (flightLeg) =>
+        AKUREYRI_FLIGHT_CODES.includes(flightLeg.origin) ||
+        AKUREYRI_FLIGHT_CODES.includes(flightLeg.destination),
     )
-    if (flightLegsLeft < flight.flightLegs.length) {
-      throw new ForbiddenException('Flight leg quota is exceeded')
+
+    if (!hasReykjavik && hasAkureyri) {
```

Comment:
It would probably be best to do
```
if (hasReykjavik) {
 countThisYears....
} else if (hasAkureyri) {
 isConnectingFlight
} else {
throw error
}
```

I was testing this and saw that it is not a requirement for non connecting flights to be from reykjavik

Reply:
I did it somewhat similarly for it wasn't as simple as that. I changed a parameter for how flights are created so that we are discoupling `discountService.useDiscount()` and `flightService.create()` semantically in the `flightController`.

`isConnectingFlight` was being used also for marking whether a flight is connectable. I replaced it with `isConnectable` for the `flightService.create()` in order to increase clarity and differentiate between intended functionality.